packages/mesh-graphql-client/test/graphql_client_test.ts

```diff
@@ -0,0 +1,424 @@
+import { getContractAddressesForChainOrThrow } from '@0x/contract-addresses';
+import { DummyERC20TokenContract } from '@0x/contracts-erc20';
+import { ExchangeContract } from '@0x/contracts-exchange';
+import { blockchainTests, constants, expect, OrderFactory, orderHashUtils } from '@0x/contracts-test-utils';
+import { BlockchainLifecycle, Web3Config, web3Factory } from '@0x/dev-utils';
+import { assetDataUtils } from '@0x/order-utils';
+import { Web3ProviderEngine } from '@0x/subproviders';
+import { DoneCallback, SignedOrder } from '@0x/types';
+import { BigNumber, hexUtils } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import { gql } from '@apollo/client';
+import 'mocha';
+
+import {
+    FilterKind,
+    OrderEvent,
+    OrderEventEndState,
+    OrderWithMetadata,
+    RejectedOrderCode,
+    SortDirection,
+} from '../src/index';
+
+import { MeshDeployment, startServerAndClientAsync } from './utils/graphql_server';
+
+blockchainTests.resets('GraphQLClient', env => {
+    describe('integration tests', () => {
+        let deployment: MeshDeployment;
+        let exchange: ExchangeContract;
+        let exchangeAddress: string;
+        let makerAddress: string;
+        let orderFactory: OrderFactory;
+        let provider: Web3ProviderEngine;
+
+        beforeEach(async () => {
+            deployment = await startServerAndClientAsync();
+        });
+
+        afterEach(async () => {
+            deployment.mesh.stopMesh();
+        });
+
+        before(async () => {
+            const chainId = await env.getChainIdAsync();
+            const accounts = await env.getAccountAddressesAsync();
+            [makerAddress] = accounts;
+
+            // Create a new provider so that the ganache instance running on port
+            // 8545 will be used instead of the in-process ganache instance.
+            const providerConfigs: Web3Config = {
+                total_accounts: constants.NUM_TEST_ACCOUNTS,
+                shouldUseInProcessGanache: false,
+                shouldAllowUnlimitedContractSize: true,
+                unlocked_accounts: [makerAddress],
+            };
+            provider = web3Factory.getRpcProvider(providerConfigs);
+
+            // HACK(jalextowle): We can't currently specify an out of process provider for a blockchainTests
+            // suite, so we need to update env.blockchainLifecycle so that the resets suite works as expected.
+            env.blockchainLifecycle = new BlockchainLifecycle(new Web3Wrapper(provider));
+
+            exchangeAddress = getContractAddressesForChainOrThrow(chainId).exchange;
+            exchange = new ExchangeContract(exchangeAddress, provider);
+            const erc20ProxyAddress = getContractAddressesForChainOrThrow(chainId).erc20Proxy;
+
+            // Configure two tokens and an order factory with a maker address so
+            // that valid orders can be created easily in the tests.
```

Comment:
I realize that this is a comment that I wrote, but I think that it would be good to specify that the addresses used to create the `makerToken` and `feeToken` are the addresses of the dummy tokens deployed in our default ganache snapshot.